extension/src/contentscript/player.ts

```diff
@@ -5,24 +5,66 @@ import {
     PlayerEvent,
 } from "../communications/from-extension-to-server";
 
+const log = require("debug")("ext:contentscript:player");
+
+export interface Event {
+    htmlEvent:
+        | "ratechange"
+        | "seeking"
+        | "progress"
+        | "ratechange"
+        | "canplaythrough"
+        | "play"
+        | "playing"
+        | "canplay"
+        | "seeked"
+        | "pause"
+        | "waiting";
+    type: MediaEventType;
+}
+
 /**
  * All the media events can be found here -> @see(https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Media_events)
  * Events we need to process via the background script
  * - playing (the `play` event is not enough, as we might have to wait for the buffer to grow after clicking play)
  * - pause (triggered on user pause, usually by clicking on the video)
  * - waiting (triggered when the buffer becomes empty and until it reaches some minimum size)
  * - seeked (triggered when a user jumps to a specific timestamp and the media finish the transition - we might want to restrict who can seek vs play/pause)
- *
- * The following events will be ignored
- * - play (because it is always followed by a playing event)
- *
- * - seeked (triggered when a user jumps to a specific timestamp - we might want to restrict who can seek vs play/pause)
+ * - canplay (triggered when the media buffer is big enough to start playing)
  */
-const events = [
+export const events: Event[] = [
+    {
+        htmlEvent: "ratechange",
+        type: MediaEventType.NOP,
+    },
+    {
+        htmlEvent: "seeking",
+        type: MediaEventType.NOP,
+    },
+    {
+        htmlEvent: "progress",
+        type: MediaEventType.NOP,
+    },
+    {
+        htmlEvent: "ratechange",
+        type: MediaEventType.NOP,
+    },
+    {
+        htmlEvent: "canplaythrough",
+        type: MediaEventType.NOP,
```

Comment:
> canplaythrough | Sent when the readyState changes to HAVE_ENOUGH_DATA, indicating that the entire media can be played without interruption, assuming the download rate remains at least at the current level. It will also be fired when playback is toggled between paused and playing. Note: Manually setting the currentTime will eventually fire a canplaythrough event in firefox. Other browsers might not fire this event.

Is this event fired in addition and/or instead of canplay ? If this event is sent only "instead of" canplay, then we need to consider it as MediaEventType.READY. If it is only "in addition", then it's fine to ignore it. 

If it can be both (ie. sometimes we can receive a canplaythrough instead of a canplay while buffering, sometimes we can receive a canplaythrough after a canplay while playing), then it's gonna be a problem : if we receive a `canplaythrough` when we're already playing, we would need to ignore this event

Reply:
I don't have all the answers but it's usually "in addition of"